File: src/pages/exercises/array-identity-matrix/_codes/js/reponse/identity-matrix.test.js
import { identityMatrix } from './identity-matrix.js';

describe('Identity Matrix', () => {
  test('it should return identity matrix of order 1', () => {
    expect(identityMatrix(1)).toEqual([[1]]);
  });

  test('it should return identity matrix of order 2', () => {
    expect(identityMatrix(2)).toEqual([
      [1, 0],
      [0, 1],
    ]);
  });

  test('it should return identity matrix of order 3', () => {
    expect(identityMatrix(3)).toEqual([
      [1, 0, 0],
      [0, 1, 0],
      [0, 0, 1],
    ]);
  });
});

File: src/pages/exercises/array-identity-matrix/_codes/python/code/identity_matrix.test.py
import unittest
from identity_matrix import identityMatrix

class Test(unittest.TestCase):
  "Identity Matrix Tool"

  def test_identity_matrix_generator(self):
    "it generates an identity matrix of order n"

    self.assertEqual(identityMatrix(2), [[1, 0], [0, 1]])
    self.assertEqual(identityMatrix(3), [[1, 0, 0], [0, 1, 0], [0, 0, 1]])
    self.assertEqual(identityMatrix(4), [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])

if (__name__ == "__main__"):
  unittest.main()